utility_analysis_new/contribution_bounders.py

```diff
@@ -13,20 +13,28 @@
 # limitations under the License.
 """ContributionBounder for utility analysis."""
 from pipeline_dp import contribution_bounders
+from pipeline_dp import sampling_utils
 
 
-class SamplingCrossAndPerPartitionContributionBounder(
+class SamplingL0LinfContributionBounder(
         contribution_bounders.ContributionBounder):
     """'Bounds' the contribution by privacy_id per and cross partitions.
 
     Because this is for Utility Analysis, it doesn't actually ensure that
     contribution bounds are enforced. Instead, it keeps track of probabilities
-    different data points are kept under per-partition and cross-partition
-    contribution bounding.
+    different data points are kept under per-partition (Linf) and
+    cross-partition (L0) contribution bounding.
+
+    If partitions_sampling_prob < 1.0, partitions are subsampled. This sampling
```

Comment:
Should we subsample on privacy IDs instead? I think that'd give a better estimate for cross-partition bounding error at the expense of a worse estimate for partition selection error. But for that we can correct by the sampling rate.

Reply:
This is very good question! Yes, we can, but it's more complicated to make it reliable for partition selection, that's why I chose to start with per partition selection. 

We can implement sampling per partition later